pkgs/swift2objc/lib/src/transformer/_core/dependencies.dart

```diff
@@ -0,0 +1,198 @@
+
+    // visit initializers
+    decl.initializers.forEach((i) => visitInitializer(i, cont));
+
+    // visit super if any
+    if (decl.superClass != null)
+      visitDeclaration(decl.superClass!.declaration, cont);
+
+    // visit nested declarations
+    decl.nestedDeclarations.forEach((n) => visitDeclaration(n, cont));
+
+    // visit protocols
+    decl.conformedProtocols.forEach((p) => visitProtocol(p.declaration, cont));
+
+    // ensure generic types do not enter
+    cont.removeWhere((t) => decl.typeParams.map((type) => type.name).contains(t));
+
+    return cont;
+  }
+
+  Set<String> visitProtocol(ProtocolDeclaration decl, [Set<String>? context]) {
+    final cont = context ??= {};
+
+    // visit variables
+    decl.properties.forEach((d) => visitVariable(d, cont));
+
+    // visit methods
+    decl.methods.forEach((m) => visitFunction(m, cont));
+
+    // visit initializers
+    decl.initializers.forEach((i) => visitInitializer(i, cont));
+
+    // visit nested declarations
+    decl.nestedDeclarations.forEach((n) => visitDeclaration(n, cont));
+
+    // visit protocols
+    decl.conformedProtocols.forEach((p) => visitProtocol(p.declaration, cont));
+
+    // ensure generic types do not enter
+    cont.removeWhere((t) => decl.typeParams.map((type) => type.name).contains(t));
+
+    return cont;
+  }
+
+  Set<String> visitInitializer(InitializerDeclaration decl,
+      [Set<String>? context]) {
+    final cont = context ??= {};
+
+    // similar to `visitMethod`, except no return type
+    decl.params.forEach((p) => visitParameter(p, cont));
+
+    return cont;
+  }
+
+  Set<String> visitFunction(FunctionDeclaration decl, [Set<String>? context]) {
+    final cont = context ??= {};
+
+    // visit parameters
+    decl.params.forEach((p) => visitParameter(p, cont));
+
+    // ensure generic types do not enter
+    cont.removeWhere((t) => decl.typeParams.map((type) => type.name).contains(t));
+
+    // visit return type
+    visitType(decl.returnType, cont);
+
+    // TODO: what of type restrictions (`... where T.Element: CustomStringConvertible`)
+
+    return cont;
+  }
+
+  Set<String> visitParameter(Parameter decl, [Set<String>? context]) {
+    final cont = context ??= {};
+
+    // just visit type of parameter
+    visitType(decl.type, cont);
+
+    return cont;
+  }
+
+  Set<String> visitVariable(VariableDeclaration decl, [Set<String>? context]) {
+    final cont = context ??= {};
+
+    // just return property type
+    visitType(decl.type, cont);
+
+    return cont;
+  }
+
+  Set<String> visitType(ReferredType type, [Set<String>? context]) {
+    final cont = context ??= {};
+
+    // we need to confirm the types located
+    // at the moment, we can perform simple regex to clean up text characters
+
+    // since we are making such visitations on normal declarations in a file,
+    // we do not need to filter out primitives at the moment
+    cont.addAll(_getTypeNames(type.swiftType));
```

Comment:
You shouldn't have to re-parse these types from strings. We did all the parsing already, and now have a nice clean AST to work with. In this case you can just check if the `type` is a `DeclaredType`, and get its `id` or `declaration` or whatever. Also need to handle the `OptionalType` case.